src/main/java/org/broadinstitute/dsm/route/KitUploadRoute.java

```diff
@@ -479,29 +497,48 @@ private int getLastNonEmptyRowIndex(String[] rows) {
-        String participantLastNameFromDoc = participantDataByFieldName.get(LAST_NAME);
+    private String userExistsInRealm(DDPInstance ddpInstanceByRealm,
+                                     Map<String, String> participantDataByFieldName) throws Exception {
+        String participantIdFromDoc = participantDataByFieldName.get(SHORT_ID).trim();
+        String participantFirstNameFromDoc = participantDataByFieldName.get(FIRST_NAME).trim().toLowerCase();
+        String participantLastNameFromDoc = participantDataByFieldName.get(LAST_NAME).trim().toLowerCase();
 
         Optional<ParticipantWrapper> maybeParticipant =
                 ParticipantWrapper.getParticipantByShortId(ddpInstanceByRealm, participantIdFromDoc);
-
-        return isKitUploadNameMatchesToEsName(participantFirstNameFromDoc, participantLastNameFromDoc, maybeParticipant);
+        maybeParticipant.orElseThrow(() -> {
+            throw new RuntimeException("Participant " + participantIdFromDoc + " does not belong to this study");
+        });
+        return checkKitUploadNameMatchesToEsName(participantFirstNameFromDoc, participantLastNameFromDoc, maybeParticipant);
     }
 
-    boolean isKitUploadNameMatchesToEsName(String participantFirstNameFromDoc, String participantLastNameFromDoc,
-                              Optional<ParticipantWrapper> maybeParticipant) {
+    String checkKitUploadNameMatchesToEsName(String participantFirstNameFromDoc, String participantLastNameFromDoc,
+                                          Optional<ParticipantWrapper> maybeParticipant) {
 
         Map<String, String> participantProfile = new HashMap<>();
         maybeParticipant.ifPresent(p -> {
             Map<String, String> participantProfileFromEs = (Map<String, String>) p.getData().get("profile");
-            participantProfile.put("firstName", participantProfileFromEs.get("firstName"));
-            participantProfile.put("lastName", participantProfileFromEs.get("lastName"));
+            participantProfile.put("firstName", participantProfileFromEs.get("firstName").trim().toLowerCase());
+            participantProfile.put("lastName", participantProfileFromEs.get("lastName").trim().toLowerCase());
         });
-        return participantFirstNameFromDoc.equals(participantProfile.get("firstName"))
-                && participantLastNameFromDoc.equals(participantProfile.get("lastName"));
+        String message = "";
+
+
+        if (!participantFirstNameFromDoc.equals(participantProfile.get("firstName"))) {
+            message += "First names ";
+            logger.error(message);
+        }
+
+        if (!participantLastNameFromDoc.equals(participantProfile.get("lastName"))) {
+            if (StringUtils.isNotBlank(message)) {
+                message += "and ";
+            }
+            message += " Last names ";
+            logger.error(message);
+        }
+        if (StringUtils.isNotBlank(message)) {
+            message += "in kit Upload don't match the participant";
```

Comment:
maybe it is better to have logger only here to have fully understandable log message.